### PR TITLE
Chore: remove build_initial_state

### PR DESCRIPTION
**Pull request by HermanObst**
## Description
Problem: all the code related with build_initial_state was useful
Solution: remove it

## Breaking changes?

- [ ] yes
- [X] no

### CODE REVIEW

File: crates/bin/prove_block/src/state_utils.rs [hunk 1]
```diff
@@ -1,153 +1,26 @@
...
-use starknet_os::starkware_utils::commitment_tree::patricia_tree::patricia_tree::PatriciaTree;
-use starknet_os::storage::storage::FactFetchingContext;
 
-use crate::rpc_utils::{CachedRpcStorage, RpcStorage};
 use crate::utils::get_subcalled_contracts_from_tx_traces;
 
 #[derive(Clone)]
 pub struct FormattedStateUpdate {
     // TODO: Use more descriptive types
+    #[allow(unused)]
```
Comment (odesenfans) [inline crates/bin/prove_block/src/state_utils.rs:14]: These are pretty suspicious. If they're unused, why do we keep them?

**Review by odesenfans — COMMENTED** [review]
One question. Good!

File: crates/bin/prove_block/src/state_utils.rs (hunk 1 above)
Comment (HermanObst) [reply crates/bin/prove_block/src/state_utils.rs:14]: This is part of the TODOs (Im going to need to touch more stuff to remove them) 
But they are also gone in the next PR :P

**Review by odesenfans — APPROVED** [review]
LGTM

**Pull request merged**